hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Listing.java

```diff
@@ -447,7 +449,7 @@ private boolean requestNextBatch() throws IOException {
      * @param objects the next object listing
      * @return true if this added any entries after filtering
      */
-    private boolean buildNextStatusBatch(S3ListResult objects) {
+    private boolean buildNextStatusBatch(S3ListResult objects) throws IOException {
```

Comment:
upate javadocs to indicate this can happen with CSE only

Reply:
ack